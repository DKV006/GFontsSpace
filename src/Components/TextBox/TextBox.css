.TextBox {
  border-color: #F9F8F3;
  padding: 60px 15px;
  background-color: rgba(226, 225, 215, 0.2); }
  .TextBox .Texbx__FontDetails {
    display: flex; }
    .TextBox .Texbx__FontDetails .FontsDetails__Item {
      color: rgba(110, 110, 60, 0.5); }
      .TextBox .Texbx__FontDetails .FontsDetails__Item:after {
        content: "|";
        display: inline-block;
        padding: 0 5px; }
      .TextBox .Texbx__FontDetails .FontsDetails__Item:hover {
        background: #FFDE00;
        color: #262626; }
        .TextBox .Texbx__FontDetails .FontsDetails__Item:hover:after {
          background: #fff; }
  .TextBox textarea {
    appearance: none;
    background: none;
    border: none;
    resize: none;
    color: rgba(110, 110, 60, 0.2);
    font-size: 80px;
    line-height: 1; }
    .TextBox textarea:hover {
      background-color: rgba(226, 225, 215, 0.2); }
    .TextBox textarea:focus {
      outline: none; }

/*# sourceMappingURL=TextBox.css.map */
