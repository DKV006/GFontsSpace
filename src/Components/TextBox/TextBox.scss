.TextBox {
  border-color: #F9F8F3;
  padding: 60px 15px;

  background-color: rgba(226,225,215,0.2); // hover color
  
  .Texbx__FontDetails {
    display: flex;
    .FontsDetails__Item {
      color: rgba(110,110,60,0.5);

      &:after {
        content: "|";
        display: inline-block;
        padding: 0 5px;
      }

      &:hover {
        background: #FFDE00;
        color: #262626;

        &:after {
          background: #fff;
        }
      }
    }
  }

  textarea {
    appearance: none;
    background: none;
    border: none;
    resize: none;
    color: rgba(110,110,60,0.2);
    font-size: 80px;
    line-height: 1;

    &:hover {
      background-color: rgba(226,225,215,0.2);
    }

    &:focus {
      outline: none;
    }
  }
}