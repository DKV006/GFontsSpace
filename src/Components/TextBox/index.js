import React, { Component } from 'react';
import PropTypes from 'prop-types';
import enhanceWithClickOutside from 'react-click-outside';
import TabContainer from './../TabContainer';
import './TextBox.css';

class TextBox extends Component {
    constructor(props) {
        super(props);
        this.state = {
            isHowToUseTabVisible: false
        }
        this.handleClickOnHowToUse = this.handleClickOnHowToUse.bind(this);
    }

    handleClickOnHowToUse() {
        let oldState = this.state.isHowToUseTabVisible
        this.setState({ isHowToUseTabVisible: !oldState })
    }

    handleClickOutside() {
        this.setState({ isHowToUseTabVisible: false })
    }

    __getFontWeight(x) {
        return x.toLowerCase() === "regular" ?  400 : x
    }

    render() {
        
        const textAreaStyle = {
            fontFamily: `'${this.props.fontName}', sans-serif`,
            fontSize: `${this.props.fontSize}px`,
            fontWeight: this.__getFontWeight(this.props.fontVariant),
            color: `rgba(${this.props.color.r}, ${this.props.color.g}, ${this.props.color.b}, ${this.props.color.a})`,
        }
        
        return (
            <div className="TextBox">
                <div className="TextBox__TextArea">
                    <ul className="Texbx__FontDetails">
                        <li className="FontsDetails__Item FontsDetails__Item--FontFamily">{this.props.fontName}</li>
                        <li className="FontsDetails__Item FontsDetails__Item--FontVariant">{this.props.fontVariant}</li>
                        <li className="FontsDetails__Item FontsDetails__Item--FontSize">{this.props.fontSize}px</li>
                    </ul>
                    <textarea style={textAreaStyle} placeholder="Write Something ..." autoFocus></textarea>
                </div>
                <div className="TextBox__CTA">
                    {/* <button type="button">
                        <i className="fa fa-plus" aria-hidden="true"></i>
                        Compare
                    </button> */}
                    <button className="HowToUse" type="button" onClick={this.handleClickOnHowToUse}>
                        <i className="fa fa-code" aria-hidden="true"></i>
                        How to use
                    </button>
                </div>
                {this.state.isHowToUseTabVisible && <TabContainer fontName={this.props.fontName} />}
            </div>
        );
    }
}

TextBox.propTypes = {

};

export default enhanceWithClickOutside(TextBox);