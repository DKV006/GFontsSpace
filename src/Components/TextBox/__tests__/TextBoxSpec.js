import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';
import TextBox from './../index';

describe('TextBox', () => {
    it('renders correctly', () => {
        const textBoxProps = {
            fontName: "Oswald",
            fontVariant: "Regular",
            fontSize: "40",
            color: {
                r: '0',
                g: '0',
                b: '0',
                a: '1',
              }
        }
        let tree = renderer.create(
            <TextBox {...textBoxProps} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
