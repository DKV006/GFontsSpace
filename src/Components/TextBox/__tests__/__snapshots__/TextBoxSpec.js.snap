// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBox renders correctly 1`] = `
<div
  className="TextBox"
>
  <div
    className="TextBox__TextArea"
  >
    <ul
      className="Texbx__FontDetails"
    >
      <li
        className="FontsDetails__Item FontsDetails__Item--FontFamily"
      >
        Oswald
      </li>
      <li
        className="FontsDetails__Item FontsDetails__Item--FontVariant"
      >
        Regular
      </li>
      <li
        className="FontsDetails__Item FontsDetails__Item--FontSize"
      >
        40
        px
      </li>
    </ul>
    <textarea
      autoFocus={true}
      placeholder="Write Something ..."
      style={
        Object {
          "color": "rgba(0, 0, 0, 1)",
          "fontFamily": "'Oswald', sans-serif",
          "fontSize": "40px",
          "fontWeight": 400,
        }
      }
    />
  </div>
  <div
    className="TextBox__CTA"
  >
    <button
      className="HowToUse"
      onClick={[Function]}
      type="button"
    >
      <i
        aria-hidden="true"
        className="fa fa-code"
      />
      How to use
    </button>
  </div>
</div>
`;

exports[`TextBox renders correctly with how to use section data 1`] = `
<div
  className="TextBox"
>
  <div
    className="TextBox__TextArea"
  >
    <ul
      className="Texbx__FontDetails"
    >
      <li
        className="FontsDetails__Item FontsDetails__Item--FontFamily"
      >
        Oswald
      </li>
      <li
        className="FontsDetails__Item FontsDetails__Item--FontVariant"
      >
        Regular
      </li>
      <li
        className="FontsDetails__Item FontsDetails__Item--FontSize"
      >
        40
        px
      </li>
    </ul>
    <textarea
      autoFocus={true}
      placeholder="Write Something ..."
      style={
        Object {
          "color": "rgba(0, 0, 0, 1)",
          "fontFamily": "'Oswald', sans-serif",
          "fontSize": "40px",
          "fontWeight": 400,
        }
      }
    />
  </div>
  <div
    className="TextBox__CTA"
  >
    <button
      className="HowToUse"
      onClick={[Function]}
      type="button"
    >
      <i
        aria-hidden="true"
        className="fa fa-code"
      />
      How to use
    </button>
  </div>
</div>
`;
